oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/cursor/TraversingCursor.java

```diff
@@ -159,10 +159,15 @@ private void fetchNext() {
 
                 readCount++;
                 if (readCount % 1000 == 0) {
-                    FilterIterators.checkReadLimit(readCount, settings);
-                    String caller = IndexUtils.getCaller(this.settings.getIgnoredClassNamesInCallTrace());
-                    LOG.warn("Traversed {} nodes with filter {} called by {}; consider creating an index or changing the query" , 
-                            readCount, filter, caller);
+                    if (readCount == 20000) {
+                        LOG.warn("Traversed {} nodes with filter {}; consider creating an index or changing the query",
+                                readCount, filter, new Exception("call stack"));
```

Comment:
Very minor suggestion: you could use Throwable instead of Exception, because it is the base class of the exception hierarchy (a matter of being minimalistic).

Reply:
Not sure if Throwable also works, according to the docs is needs to be an "exception" object: https://www.slf4j.org/faq.html#paramException

Reply:
They are being imprecise with the use of the term exception, the method takes a Throwable as an argument:
https://www.javadoc.io/doc/org.slf4j/slf4j-api/latest/org/slf4j/Logger.html#warn(java.lang.String,java.lang.Throwable)